src/midi/midi.cpp

```diff
@@ -801,97 +794,95 @@ void init_midi_dosbox_settings(Section_prop& secprop)
+	                        "               for detailed configuration.");
+
+	str_prop->SetOptionHelp(DefaultMidiDevicePref,
+	                        "  port:        A MIDI port of the host operating system's MIDI interface\n"
+	                        "               (default). The MIDI port to use can be configured with the\n"
+	                        "               'midiconfig' setting.");
+
 	str_prop->SetOptionHelp("none", "  none:        Disable MIDI output.");
 	str_prop->Set_values({
-		"auto",
-#if defined(MACOSX)
-#if C_COREMIDI
-		        MidiDeviceName::CoreMidi,
-#endif
+		MidiDevicePortPref,
 #if C_COREAUDIO
 		        MidiDeviceName::CoreAudio,
 #endif
-#elif defined(WIN32)
-		        MidiDeviceName::Win32,
-#endif
-#if C_ALSA
-		        MidiDeviceName::Alsa,
-#endif
 #if C_FLUIDSYNTH
 		        MidiDeviceName::FluidSynth,
 #endif
 		        MidiDeviceName::Mt32, "none"
 	});
 
+	str_prop->SetDeprecatedWithAlternateValue("win32", DefaultMidiDevicePref);
+	str_prop->SetDeprecatedWithAlternateValue("coremidi", DefaultMidiDevicePref);
+	str_prop->SetDeprecatedWithAlternateValue("alsa", DefaultMidiDevicePref);
+	str_prop->SetDeprecatedWithAlternateValue("oss", DefaultMidiDevicePref);
+	str_prop->SetDeprecatedWithAlternateValue("auto", DefaultMidiDevicePref);
+
 	str_prop = secprop.Add_string("midiconfig", WhenIdle, "");
 	str_prop->Set_help(
-	        "Configuration options for the selected MIDI interface (unset by default).\n"
-	        "This is usually the ID or name of the MIDI synthesizer you want\n"
-	        "to use (find the ID/name with the DOS command 'MIXER /LISTMIDI').\n"
+	        "Configuration options for the selected MIDI device (unset by default).\n"
 	        "Notes:");
 
 	str_prop->SetOptionHelp("fluidsynth_or_mt32emu",
-	                        "  - This option has no effect when using the built-in synthesizers\n"
+	                        "  - The setting has no effect when using the built-in synthesizers\n"
 	                        "    ('mididevice = fluidsynth' or 'mididevice = mt32').");
 
-	str_prop->SetOptionHelp("coreaudio",
-	                        "  - When using 'coreaudio', you can specify a SoundFont here.");
+	str_prop->SetOptionHelp(
+	        "coreaudio",
+	        "  - When using 'mididevice = coreaudio', this setting specifies the SoundFont\n"
+	        "    to use. You must use the absolute path of the SoundFont file.");
 
-	str_prop->SetOptionHelp("alsa",
-	                        "  - When using ALSA, use the Linux command 'aconnect -l' to list all open\n"
-	                        "    MIDI ports and select one (e.g. 'midiconfig = 14:0' for sequencer\n"
+	str_prop->SetOptionHelp("linux",
+	                        "  - When using 'mididevice = port', use the Linux command 'aconnect -l' to list\n"
+	                        "    all open MIDI ports and select one (e.g., 'midiconfig = 14:0' for sequencer\n"
```

Comment:
I would suggest leaving this config option named ALSA.  Changing this breaks backwards compatibility with old config files and is less clear as to the interface we're using.  There are other MIDI interfaces on Linux such as JACK.  It's also inconsistent with what we're doing on Mac (the config option is named "coreaudio" not "macosx".)

Reply:
Reflecting to your concrete points first @weirddan455:

> I would suggest leaving this config option named ALSA. Changing this breaks backwards compatibility with old config files and is less clear as to the interface we're using.

No, it does not break backwards compatibility. If you set `mididevice = alsa`, that would revert to `mididevice = port` via the deprecation mechanism. Also, you can set it to `mididevice = blah` too, which would revert to `port`. Try it and see! 😄 

> There are other MIDI interfaces on Linux such as JACK.

AFAIK, JACK is built on top of ALSA. Virtually zero normal users would use JACK as it's a complicated beast. I used it one (on macOS, though), and yeah it was pretty horrible. It seems it's universally hated:

https://www.reddit.com/r/linuxaudio/comments/gmjl9s/jack_and_its_remarkable_ability_to_kill_creativity/
https://linuxmusicians.com/viewtopic.php?t=19258
https://linuxmusicians.com/viewtopic.php?t=24344
https://linuxmusicians.com/viewtopic.php?t=25557

To me, this means ALSA is the de-facto MIDI interface on Linux, and it's extremely unlikely we'll ever add JACK support, especially because they're not mutually exclusive. ALSA would always work for JACK users.

> It's also inconsistent with what we're doing on Mac (the config option is named "coreaudio" not "macosx".)

No, read this section more carefully. `linux`, `windows_or_macos`, `mt32` here are _not_ actual setting names but just some symbolic constants you pass into `SetEnabledOptions`. This is a recent feature that enables us conditionally include just some of these OS-specific help texts with define macros, _but_ they will all show up in the translations because they're all added with `SetOptionHelp`. 

---

Okay, so my general comments. Please read my original description carefully and check out the screenshots that show the settings' help because I have a feeling you haven't fully understood what I'm doing here.

Like I said, we have a one-to-one mapping of MIDI interfaces and operating systems now. As you've pointed out, JACK could break this, at least at the first glance, but I described above, I don't think that's the case.

I'm not dying to get the new `port` setting in, but I thought it's neat and better than the vague `auto` setting.

If you think having `alsa` and `jack` (in the future) for Linux is important, we could rename `port` back to `auto`, then `auto` would mean `alsa` on Linux, or you could set `alsa` or `jack` explicitly. But the `win32` and `coremidi` options would go away; there's no other way for system-level MIDI on Windows and macOS so `auto` could always mean only one thing.

I just don't really like `auto` when what it really means is "use the system-level MIDI interface".

Reply:
Oh, I missed that "linux" here was just an internal name.  I built it and checked how the config looks and it looks good.  Agree about auto being confusing.  We can always change internal naming later if/when needed.